apps/signal/56-forms-and-signal/src/app/order.component.ts

```diff
@@ -55,18 +57,22 @@ import { products } from './products';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export default class OrderComponent {
+  private readonly route = inject(ActivatedRoute);
   form = new FormGroup({
-    quantity: new FormControl(1, { nonNullable: true }),
+    quantity: new FormControl(
+      +(this.route.snapshot.queryParams['quantity'] ?? 1),
```

Comment:
this is the secret. 
activatedRoute is initialized before component initialization, so this works